### PR TITLE
Aggregate logs in `evaluate`

### LINKED ISSUE #481: BasicWorkflow: Unstable val_losses due to validation data shuffling
In the `BasicWorkflow` class, the validation set is constructed via `validation_data = OfflineDataset(data=validation_data, batch_size=dataset.batch_size, adapter=self.adapter)` [in the `_fit` method](https://github.com/bayesflow-org/bayesflow/blob/main/bayesflow/workflows/basic_workflow.py#L882). This treats the validation data as a standard `OfflineDataset` and implies shuffling on epoch end. Thus, stochasticity is introduced by modifying the composition of the evaluated batches, which can lead to substantial changes between multiple validation loss calculations during training even for frozen networks. 

E.g., adapting Linear_Regression_Starter.ipynb to use offline training with a learning rate of 0 (and no standardization in the adapter),

```python
validation_data = workflow.simulate(1000)
history = workflow.fit_offline(
    data=training_data,
    epochs=5,
    batch_size=32,
    validation_data=validation_data,
)
```

leads to highly variable validation losses:

```python
[19.465839385986328,
 10.662302017211914,
 16.612285614013672,
 7.247644901275635,
 17.282154083251953]
```

Setting the number of validation sets equal the batch size, either via `validation_data = workflow.simulate(32)` or via `batch_size=1000`, enforces equal batches despite shuffling, resulting in stable validation losses:

```python
[12.364879608154297,
 12.364879608154297,
 12.364879608154297,
 12.364879608154297,
 12.364879608154297]
```

A pull request with a simple fix will follow.

### PR DESCRIPTION
**Pull request by LarsKue**
Fixes #481

### CODE REVIEW

File: bayesflow/approximators/backend_approximators/torch_approximator.py [hunk 1]
```diff
@@ -3,13 +3,110 @@
...
+        y=None,
+        batch_size=None,
+        verbose="auto",
+        sample_weight=None,
+        steps=None,
+        callbacks=None,
+        return_dict=False,
+        **kwargs,
+    ):
+        # TODO: respect compiled trainable state
```
[the remark this answers is not in the document]
Comment (LarsKue) [reply bayesflow/approximators/backend_approximators/torch_approximator.py:43]: This is code from keras, so we won't fix this.

File: bayesflow/approximators/backend_approximators/torch_approximator.py [hunk 2]
```diff
@@ -3,13 +3,110 @@
 
 from bayesflow.utils import filter_kwargs
 
+from keras.src.backend.torch.trainer import TorchEpochIterator
+from keras.src import callbacks as callbacks_module
+
 
 class TorchApproximator(keras.Model):
```
[the remark this answers is not in the document]
Comment (LarsKue) [reply bayesflow/approximators/backend_approximators/torch_approximator.py:10]: Code duplication should be preferred here.

File: bayesflow/approximators/backend_approximators/jax_approximator.py [hunk 3]
```diff
@@ -3,13 +3,148 @@
...
+
+        # BAYESFLOW: average the metrics across all iterations
+        logs = self._mean_logs(logs, total_steps)
+
+        # Reattach state back to model (if not already done by a callback).
+        self.jax_state_sync()
+
+        # The jax spmd_mode is need for multi-process context, since the
+        # metrics values are replicated, and we don't want to do a all
+        # gather, and only need the local copy of the value.
```
[the remark this answers is not in the document]
Comment (LarsKue) [reply bayesflow/approximators/backend_approximators/jax_approximator.py:136]: Copied from keras, so we will leave this unmodified

**Comment by LarsKue** [comment]
Superceded by #485
